build.sbt

```
@@ -325,61 +327,35 @@ lazy val connectClient = (project in file("spark-connect/client"))
     ),
-    (Test / javaOptions) += s"-Ddelta.test.home=" + file(".").getAbsoluteFile.getParentFile,
-    (Test / resourceGenerators) += Def.task {
-      val location = url("https://dist.apache.org/repos/dist/release/spark/spark-4.0.0-preview1/spark-4.0.0-preview1-bin-hadoop3.tgz")
-      val destDir = (Test / resourceManaged).value / "spark"
-      if (!destDir.exists()) {
-        IO.createDirectory(destDir)
-        val files = mutable.Buffer.empty[File]
-        Using(new BufferedInputStream(location.openStream())) { bi =>
-          Using(new GzipCompressorInputStream(bi)) { gi =>
-            Using(new TarArchiveInputStream(gi)) { ti =>
-              var entry = ti.getNextEntry
-              while (entry != null) {
-                val dest = destDir / entry.getName
-                if (entry.isDirectory) {
-                  dest.mkdirs()
-                } else {
-                  files += dest
-                  Using(Files.newOutputStream(dest.toPath)) { os =>
-                    IOUtils.copy(ti, os)
-                  }
-
-                  val perms = new util.HashSet[PosixFilePermission]()
-                  perms.add(PosixFilePermission.OWNER_EXECUTE)
-                  perms.add(PosixFilePermission.OWNER_READ)
-                  perms.add(PosixFilePermission.OWNER_WRITE)
-                  Files.setPosixFilePermissions(dest.toPath, perms)
-                }
-                entry = ti.getNextEntry
-              }
-            }
-          }
+    (Test / javaOptions) += {
+      // Create a (mini) Spark Distribution based on the server classpath.
+      val serverClassPath = (connectServer / Compile / fullClasspath).value
+      val distributionDir = crossTarget.value / "test-dist"
+      if (!distributionDir.exists()) {
+        val jarsDir = distributionDir / "jars"
+        IO.createDirectory(jarsDir)
+        // Create symlinks for all dependencies.
+        serverClassPath.distinct.foreach { entry =>
+          val jarFile = entry.data.toPath
+          val linkedJarFile = jarsDir / entry.data.getName
+          Files.createSymbolicLink(linkedJarFile.toPath, jarFile)
         }
-        files
-      } else {
-        destDir.get()
-      }
-    }.taskValue,
-    (Test / resourceGenerators) += Def.task {
-      val src = url("https://repository.apache.org/content/groups/public/org/apache/spark/spark-connect_2.13/4.0.0-preview1/spark-connect_2.13-4.0.0-preview1.jar")
-      val dest = (Test / resourceManaged).value / "spark-connect.jar"
-      if (!dest.exists()) {
-        src #> dest !;
-        Seq(dest)
-      } else {
-        dest.get()
+        // Create a symlink for the log4j properties
+        val confDir = distributionDir / "conf"
+        IO.createDirectory(confDir)
+        val log4jProps = (spark / Test / resourceDirectory).value / "log4j2_spark_master.properties"
+        val linkedLog4jProps = confDir / "log4j2.properties"
+        Files.createSymbolicLink(linkedLog4jProps.toPath, log4jProps.toPath)
```

Nit: A way to add named arguments to Java like Scala
```suggestion
        Files.createSymbolicLink(linkedLog4jProps.toPath /* link */, log4jProps.toPath /* target */)
```